File: ios/IceBloxApp/Persistence/OfflineQueueEntry.swift
import Foundation

struct OfflineQueueEntry: Codable {
    let id: Int64?
    let plateHash: String
    let timestamp: Date
    let latitude: Double?
    let longitude: Double?
    let sessionID: String
    let confidence: Float
    let isPrimary: Bool

    init(
        plateHash: String,
        timestamp: Date = Date(),
        latitude: Double? = nil,
        longitude: Double? = nil,
        sessionID: String,
        confidence: Float = 0,
        isPrimary: Bool = false
    ) {
        self.id = nil
        self.plateHash = plateHash
        self.timestamp = timestamp
        self.latitude = latitude
        self.longitude = longitude
        self.sessionID = sessionID
        self.confidence = confidence
        self.isPrimary = isPrimary
    }

    // swiftlint:disable:next line_length
    init(id: Int64, plateHash: String, timestamp: Date, latitude: Double?, longitude: Double?, sessionID: String, confidence: Float = 0, isPrimary: Bool = false) {
        self.id = id
        self.plateHash = plateHash
        self.timestamp = timestamp
        self.latitude = latitude
        self.longitude = longitude
        self.sessionID = sessionID
        self.confidence = confidence
        self.isPrimary = isPrimary
    }
}

File: android/app/src/main/java/com/iceblox/app/persistence/OfflineQueueEntry.kt
package com.iceblox.app.persistence

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "offline_queue")
data class OfflineQueueEntry(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    @ColumnInfo(name = "plate_hash") val plateHash: String,
    @ColumnInfo(name = "timestamp") val timestamp: Long,
    @ColumnInfo(name = "latitude") val latitude: Double?,
    @ColumnInfo(name = "longitude") val longitude: Double?,
    @ColumnInfo(name = "session_id", defaultValue = "") val sessionId: String,
    @ColumnInfo(name = "confidence", defaultValue = "0") val confidence: Float = 0f,
    @ColumnInfo(name = "is_primary", defaultValue = "0") val isPrimary: Boolean = false
)
